IANotificationsTestProject/In App Notifications/NoticeController.swift

```diff
@@ -0,0 +1,65 @@
+import UIKit
+
+class NoticeController {
+    private var notices: [Notice] = []
+    private let noticePresenter = NoticePresenter()
+
+    private var activeViewIsBeingTouched: Bool = false
+
+    func present(_ notice: Notice) {
+        if noticePresenter.isPresenting {
+            notices.append(notice)
+        } else {
```

Comment:
Nipticky (no need to actually change anything!) You can improve readability by doing something like this

```
if noticePresenter.isPresenting {
   // ...
   return
}
```

(And drop the else!)